src/main/java/uk/gov/pay/connector/charge/util/ChargeCreateRequestIdempotencyComparatorUtil.java

```diff
@@ -0,0 +1,18 @@
+package uk.gov.pay.connector.charge.util;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.gov.pay.connector.charge.model.ChargeCreateRequest;
+
+import java.util.Map;
+
+public class ChargeCreateRequestIdempotencyComparatorUtil {
+    private static final ObjectMapper mapper = new ObjectMapper();
+    public static boolean compare(ChargeCreateRequest request, Map<String, Object> idempotencyRequestBody) {
+        JsonNode requestNode = mapper.valueToTree(request);
+        ChargeCreateRequest idempotencyRequest = new ObjectMapper().convertValue(idempotencyRequestBody, ChargeCreateRequest.class);
+        JsonNode idempotencyNode = mapper.valueToTree(idempotencyRequest);
```

Comment:
```suggestion
        JsonNode idempotencyNode = mapper.valueToTree(idempotencyRequestBody);
```
I think we can just do this directly from the map rather than having to read it into a ChargeCreateRequest first